tests/test_timeable_mixin.py

```
@@ -24,69 +23,84 @@
+    T = TimeableDerived()
 
-    def test_context_manager(self):
-        T = TimeableDerived()
+    benchmark(T.decorated_takes_time, 0.00001)
 
-        T.uses_contextlib(num_seconds=1)
 
-        duration = T._times_for("using_contextlib")[-1]
-        np.testing.assert_almost_equal(duration, 1, decimal=1)
+def test_pprint_num_unit():
+    assert (5, "μs") == TimeableMixin._get_pprint_num_unit(5 * 1e-6)
```

Ruff

Yoda condition detected

(SIM300)